test/extensions/transport_sockets/tls/ecdsa_private_key_method_provider.cc

```diff
@@ -0,0 +1,176 @@
+#include "test/extensions/transport_sockets/tls/ecdsa_private_key_method_provider.h"
```

Comment:
I know that those are test providers, but we really don't need to have separate providers for ECDSA & RSA, this should be the same provider that can handle both types, since the differences are minimal.

This will also uncover issues with using static index to register private key connection on the `SSL` object.

Reply:
Ok, I'll merge the providers.